### DISCUSSION TITLE
How does Viper store data of the flags?

### DISCUSSION THREAD

**Discussion by MichelDiz** [spf13/viper#1455] [Q&A] [answered]
I notice that sometimes when I change a flag value, it doesn't seem to override it.
Ps; it could be an application problem (a very complex open source project).

My question boils down to knowing how it is stored.

In the memory?
On disk by some kind of binary?
Any other?

Thanks!

**Comment by sagikazarmark** [comment, marked as answer]
Well, Viper doesn't store them per se. It reads them from the argument list and keeps it in memory.

Not sure what override means in this case. Can you give a concrete, reproducible example?

**Reply by MichelDiz** [reply]
Override in the sense of overwriting. When I restart an application with another value in that flag.

But now I'm suspecting something else. Maybe the Viper isn't the problem. Maybe it's the implementation of the RAFT Algorithm that we use. Somewhere it does not change the data correctly.

The information that goes into memory was helpful. So it's an ephemeral value. Right?

**Reply by sagikazarmark** [reply]
It is, Viper doesn't persist anything (unless you tell it to write something to a file). Flags generally take precedence over everything. It's probably trivial to verify that the config is right, so I'd start there.
